src/bit_machine/mod.rs

```diff
@@ -365,67 +369,74 @@ impl BitMachine {
         }
     }
 
-    fn exec_jet<J: Jet>(&mut self, jet: J, env: &J::Environment) -> Result<(), JetFailed> {
-        use simplicity_sys::c_jets::frame_ffi::{c_readBit, c_writeBit, CFrameItem};
-        use simplicity_sys::c_jets::uword_width;
-        use simplicity_sys::ffi::UWORD;
+    /// Return a read frame that is equivalent to the active read frame.
+    fn get_input_frame(&mut self, bit_width: usize, buffer: &mut Vec<UWORD>) -> CFrameItem {
```

Comment:
In 63c2d233fb29a8837e930d79507d89d06a4443d9:

`get_input_frame` and `get_output_frame` both need to be marked `unsafe` since they return a data structure containing raw pointers into a `Vec`, with no protection on those pointers being invalidated.

Probably these methods should just be defined inside of `exec_jet` since they're so wildly unsafe to use.

Reply:
I moved the new functions inside `exec_jet`. Because these functions are pure, we need to pass `self` as a parameter. I think this is a better solution than closures which cannot be unsafe.